src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Using a cached thread pool can potentially create a large number of threads if many events are being notified concurrently. This could lead to performance issues. Consider using a fixed thread pool or another type of ExecutorService that limits the number of threads.